sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -553,6 +557,8 @@ object DataSource extends Logging {
       "org.apache.spark.sql.execution.datasources.parquet.DefaultSource" -> parquet,
       "org.apache.spark.sql.hive.orc.DefaultSource" -> orc,
       "org.apache.spark.sql.hive.orc" -> orc,
+      "org.apache.spark.sql.execution.datasources.orc.DefaultSource" -> nativeOrc,
+      "org.apache.spark.sql.execution.datasources.orc" -> nativeOrc,
```

Comment:
This map is for backward compatibility in case we move data sources around. I think this `datasources.orc` is newly added. Why we need to add them here?

Reply:
ah good catch! sounds like we don't need compatibility rule for the new orc.

Reply:
Like `USING org.apache.spark.sql.hive.orc`, we want to use `USING org.apache.spark.sql.execution.datasources.orc`, don't we?

Reply:
When I received [the advice](https://github.com/apache/spark/pull/19871#discussion_r154580007), I thought it's for consistency.

Reply:
This is for safety. We also do it for parquet

Reply:
For parquet, this is for historical reason, see #13311.

Previously you can use parquet by `org.apache.spark.sql.execution.datasources.parquet` and `org.apache.spark.sql.execution.datasources.parquet.DefaultSource`. So it is also for backward compatibility.

For this new orc, it is not the same case.

Reply:
I think we should rename variable and/or fix the comments there when we touch some codes around there to prevent confusion next time though.

Reply:
`org.apache.spark.sql.execution.datasources.*` path is meant to be private too. But I think it's okay to leave it for practical use cases with some comments.

Reply:
These changes are pretty safe. In case, we move the orc to the other location, it will still refer to the right location.